test/correctness/simd_op_check_hvx.cpp

```diff
@@ -225,10 +225,18 @@ class SimdOpCheckHVX : public SimdOpCheckTest {
         check("vpacko(v*.h,v*.h)", hvx_width/1, u8(i16_1 >> 8));
         check("vpacko(v*.h,v*.h)", hvx_width/1, i8(u16_1 >> 8));
         check("vpacko(v*.h,v*.h)", hvx_width/1, i8(i16_1 >> 8));
-        check("vpacko(v*.w,v*.w)", hvx_width/2, u16(u32_1 >> 16));
-        check("vpacko(v*.w,v*.w)", hvx_width/2, u16(i32_1 >> 16));
-        check("vpacko(v*.w,v*.w)", hvx_width/2, i16(u32_1 >> 16));
-        check("vpacko(v*.w,v*.w)", hvx_width/2, i16(i32_1 >> 16));
+        check("vpacko(v*.w,v*.w)", hvx_width/2, u16(u32_1 >> cast<unsigned>(16)));
```

Comment:
I think the ones that need testing are the vector RHS tests above (vasl, vasr, vlsr). Your change only affects vector code (as it should), so it might be best to revert the changes to these shifts.

Reply:
These have been reverted, thanks.